pkg/agent/cniserver/pod_configuration_test.go

```diff
@@ -0,0 +1,461 @@
+			PortNo: 1,
+			Length: 72,
+			Name:   []byte(fmt.Sprintf("%s\x00", podIfName)),
+			State:  openflow15.PS_LIVE,
+		},
+	}
+
+	podIfaceConfig = interfacestore.InterfaceConfig{
+		InterfaceName: podIfName,
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+			PodNamespace: testPodNamespace,
+			PodName:      testPodNameA,
+			ContainerID:  podInfraContainerID,
+		},
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: "test-port-uuid",
+		},
+		IPs: podIPs,
+		MAC: podMac,
+	}
+)
+
+type mockClients struct {
+	kubeClient       *fakeclientset.Clientset
+	informerFactory  informers.SharedInformerFactory
+	localPodInformer cache.SharedIndexInformer
+	podLister        corelisters.PodLister
+	ofClient         *openflowtest.MockClient
+	recorder         *record.FakeRecorder
+}
+
+func newMockClients(ctrl *gomock.Controller, nodeName string, objects ...runtime.Object) *mockClients {
+	kubeClient := fakeclientset.NewClientset()
+	for _, obj := range objects {
+		if obj != nil {
+			kubeClient.Tracker().Add(obj)
+		}
+	}
+	informerFactory := informers.NewSharedInformerFactory(kubeClient, 0)
+
+	localPodInformer := coreinformers.NewFilteredPodInformer(
+		kubeClient,
+		metav1.NamespaceAll,
+		0,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		func(options *metav1.ListOptions) {
+			options.FieldSelector = fields.OneTermEqualSelector("spec.nodeName", nodeName).String()
+		},
+	)
+	podLister := corelisters.NewPodLister(localPodInformer.GetIndexer())
+	ofClient := openflowtest.NewMockClient(ctrl)
+	recorder := record.NewFakeRecorder(100)
+	recorder.IncludeObject = false
+
+	return &mockClients{
+		kubeClient:       kubeClient,
+		informerFactory:  informerFactory,
+		localPodInformer: localPodInformer,
+		podLister:        podLister,
+		ofClient:         ofClient,
+		recorder:         recorder,
+	}
+}
+
+func (c *mockClients) startInformers(stopCh chan struct{}) {
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+	go c.localPodInformer.Run(stopCh)
+	cache.WaitForCacheSync(stopCh, c.localPodInformer.HasSynced)
+}
+
+type asyncWaiter struct {
+	podName     string
+	containerID string
+	waitCh      chan struct{}
+	stopCh      chan struct{}
+	notifier    *channel.SubscribableChannel
+}
+
+func (w *asyncWaiter) notify(e interface{}) {
+	podUpdate := e.(types.PodUpdate)
+	if podUpdate.PodName == w.podName && podUpdate.ContainerID == w.containerID {
+		w.waitCh <- struct{}{}
+	}
+}
+
+func (w *asyncWaiter) wait() {
+	<-w.waitCh
+}
+
+func (w *asyncWaiter) close() {
+	close(w.waitCh)
+	close(w.stopCh)
+}
+
+func newAsyncWaiter(podName, containerID string) *asyncWaiter {
+	waiter := &asyncWaiter{
+		podName:     podName,
+		containerID: containerID,
+		waitCh:      make(chan struct{}),
+		stopCh:      make(chan struct{}),
+		notifier:    channel.NewSubscribableChannel("PodUpdate", 100),
+	}
+	waiter.notifier.Subscribe(waiter.notify)
+	go waiter.notifier.Run(waiter.stopCh)
```

Comment:
The `stopCh` should come from the caller and doesn't need to be stored in `asyncWriter`